src/Analyzers/CSharp/Tests/UseImplicitObjectCreation/CSharpUseImplicitObjectCreationTests.cs

```diff
@@ -662,6 +662,207 @@ void M()
     {
         List<(int SomeName, int SomeOtherName, int YetAnotherName)> list = new();
     }
+}",
+                LanguageVersion = CodeAnalysis.CSharp.LanguageVersion.CSharp9,
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseImplicitObjectCreation)]
+        public async Task TestWithArrayInitializerExpression()
+        {
+            await new VerifyCS.Test
+            {
+                TestCode = @"
+class C
+{
+    void bar()
+    {
+        var a = new C[]
+        {
+            new [|C|](),
+        };
+    }
+}",
```

Comment:
Can you add a test similar to:

```
class C
{
    void bar()
    {
        var a = new[]
        {
            new C(),
        };
    }
}
```

Reply:
I don't think that's necessary. Matches on outer parent are always on `ArrayCreationExpressionSyntax`/`ObjectCreationExpressionSyntax`, never on an implicit expression, as that won't work by definition.